Allow usage of Quarkus Gradle Plugin as a custom plugin dependency

**Description**
It would be nice to hide some default configuration for our used quarkus version/dependecies/etc behind our own custom Gradle plugin. This is useful, because we have multiple quarkus apps managed by our team that are developed and released together; as such, we would like to be able to define the used quarkus version and some settings in a central place; i.e. a gradle plugin and not having to define the quarkus gradle plugin dependency in each project individually.

This custom gradle plugin should dependent on the quarkus gradle plugin in it's build-script to pull it in transitively:
```
implementation 'io.quarkus:io.quarkus.gradle.plugin:1.8.1.Final'
```

Our custom gradle plugin will now apply the quarkus gradle plugin:

```
  @Override
    public void apply(Project project) {
		project.apply plugin: 'io.quarkus'
                // add some default dependencies, etc....
    }
```

There should be NO direct dependency in the settings.gradle of the quarkus project like this (all should be hidden by our custom gradle plugin if possible):

```
pluginManagement {
    plugins {
      id 'io.quarkus' version "1.8.1.Final"
    }
}
```

Unfortunately, when applying this plugin and pulling it in just as a transitive dependency, the quarkusDev plugin will fail because it searches for the quarkus gradle plugin dependency just in the direct dependencies.
(https://github.com/quarkusio/quarkus/blob/master/devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java#L489)

```
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: java.lang.IllegalStateException: Unable to find quarkus-gradle-plugin dependency in project ':quarkus-minimal-assembly'
        at io.quarkus.gradle.tasks.QuarkusDev.addGradlePluginDeps(QuarkusDev.java:529)
        at io.quarkus.gradle.tasks.QuarkusDev.startDev(QuarkusDev.java:314)
        ... 92 more
```


**Implementation ideas**
For such use-cases, it would be good if the quarkus gradle plugin is also search in the transitive dependencies; which might be the case if it is just pulled in by some other plugin.

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java
@@ -481,6 +481,23 @@ public class QuarkusDev extends QuarkusTask {
                 .ofNullable((JavaCompile) getProject().getTasks().getByName(JavaPlugin.COMPILE_JAVA_TASK_NAME));
     }
 
+    private ResolvedDependency findQuarkusPluginDependency(Set<ResolvedDependency> dependencies) {
+        for (ResolvedDependency rd : dependencies) {
+            if ("io.quarkus.gradle.plugin".equals(rd.getModuleName())) {
+                return rd;
+            } else {
+                Set<ResolvedDependency> children = rd.getChildren();
+                if (children != null) {
+                    ResolvedDependency quarkusPluginDependency = findQuarkusPluginDependency(children);
+                    if (quarkusPluginDependency != null) {
+                        return quarkusPluginDependency;
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
     private void addGradlePluginDeps(StringBuilder classPathManifest, DevModeContext context) {
         boolean foundQuarkusPlugin = false;
         Project prj = getProject();
@@ -491,14 +508,15 @@ public class QuarkusDev extends QuarkusTask {
             if (firstLevelDeps.isEmpty()) {
                 // TODO this looks weird
             } else {
-                for (ResolvedDependency rd : firstLevelDeps) {
-                    if ("io.quarkus.gradle.plugin".equals(rd.getModuleName())) {
-                        rd.getAllModuleArtifacts().stream()
-                                .map(ResolvedArtifact::getFile)
-                                .forEach(f -> addToClassPaths(classPathManifest, f));
-                        foundQuarkusPlugin = true;
-                        break;
-                    }
+                ResolvedDependency quarkusPluginDependency = findQuarkusPluginDependency(firstLevelDeps);
+                if (quarkusPluginDependency != null) {
+                    quarkusPluginDependency.getAllModuleArtifacts().stream()
+                            .map(ResolvedArtifact::getFile)
+                            .forEach(f -> addToClassPaths(classPathManifest, f));
+
+                    foundQuarkusPlugin = true;
+
+                    break;
                 }
             }
             prj = prj.getParent();

